nox/sessions.py

```diff
@@ -405,6 +405,10 @@ def run(
             msg = "First argument to `session.run` is a list. Did you mean to use `session.run(*args)`?"
             raise ValueError(msg)
 
+        if self._runner.global_config.dry_run:
```

Comment:
Shouldn't these be moved down? It would not run {args} even if --dry-run was not set, because --install-only was set too.